app/src/main/java/org/mozilla/fenix/settings/WallpaperSettingsFragment.kt

```diff
@@ -0,0 +1,175 @@
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.background
+import androidx.compose.foundation.border
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.aspectRatio
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.lazy.GridCells
+import androidx.compose.foundation.lazy.items
+import androidx.compose.foundation.lazy.LazyVerticalGrid
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.Surface
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.ViewCompositionStrategy
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.fragment.app.Fragment
+import org.mozilla.fenix.R
+import org.mozilla.fenix.databinding.FragmentWallpaperSettingsBinding
+import org.mozilla.fenix.ext.requireComponents
+import org.mozilla.fenix.theme.FirefoxTheme
+import org.mozilla.fenix.wallpapers.Wallpaper
+import org.mozilla.fenix.wallpapers.WallpaperManager
+
+class WallpaperSettingsFragment : Fragment() {
+    private val binding by lazy {
+        FragmentWallpaperSettingsBinding.inflate(layoutInflater)
+    }
+
+    private val wallpaperManager by lazy {
+        requireComponents.wallpaperManager
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        binding.composeView.apply {
+            setViewCompositionStrategy(ViewCompositionStrategy.DisposeOnViewTreeLifecycleDestroyed)
+            setContent {
+                FirefoxTheme {
+                    WallpaperSettings(wallpaperManager)
+                }
+            }
+        }
+        return binding.root
+    }
+}
+
+@Composable
+fun WallpaperSettings(
+    wallpaperManager: WallpaperManager
+) {
+    val currentlySelectedWallpaper = wallpaperManager.currentWallpaper.collectAsState()
+    Column {
+        Surface {
+            WallpaperThumbnails(
+                wallpapers = wallpaperManager.wallpapers.toList(),
+                onSelectionChanged = { wallpaperManager.updateWallpaperSelection(it) },
+                selectedWallpaper = currentlySelectedWallpaper.value
+            )
+        }
+    }
+}
+
+@OptIn(ExperimentalFoundationApi::class)
+@Composable
+fun WallpaperThumbnails(
+    wallpapers: List<Wallpaper>,
+    onSelectionChanged: (Wallpaper) -> Unit,
```

Comment:
I have a slight preference for keeping these types of lambdas as the bottom parameter items. This is mostly coming from my React background. 

WallpaperThumbnails(
    wallpapers: List<Wallpaper>,
    onSelectionChanged: (Wallpaper) -> Unit,
 

 
I have a slight preference for keeping these types of lambdas as the bottom parameter items. This is mostly coming from my React background. 

Ultimately, I think about the flow when I am reading the code, it would easier to see the simple parameters (Int, String, etc) first and whereas providing a lambda function disrupt the readability of things.

Here's a visual example:
```
WallpaperThumbnails(
    wallpapers = wallpapers,
    selectedWallpaper = selectedWallpaper,
    numColumns = 4,
    onSelectionChanged = { wallpaper ->
        someFunction(wallpaper)
        anotherCallback(wallpaper)
   }
)
```

Reply:
This was actually very intentional. Here's my reasoning:

Jetpack Compose has defined the convention that `@Composable` slots should generally be the final parameter to other composable functions. The reason being that we get the nice visual cues for parent-child relationships in our view tree:
```
@Composable
fun Tree() {
    Surface {
        Button {
            Text()
...
```

If we have trailing lambdas that aren't composables but are visually within a compose scope, that could be also be pretty hostile to the reader:
```
@Composable
fun Tree() {
    Surface {
        WallpaperThumbnails {
            // now we're in the onSelectionChanged lambda
```

I agree that having the lambda types in the middle is pretty gross, but my method allows us to rely on our tools to avoid situations like this completely. I wonder if maybe there's a convention we could come up with that's a compromise, such as defining the lambda outside constructor call as to not interrupt the parameter list:

@Composable
fun Tree() {
    Surface {
        val onWallpaperSelectionChanged: (Wallpaper) -> Unit = {}
        WallpaperThumbnails(
            onSelectionChanged = onWallpaperSelectionChanged
...
```

Reply:
I did revise my comment to add the exception about the @Composable content after submitting it initially. In this case, I was not really thinking about the middle case you presented which is that we use trailing lambdas - which is a pattern I am hoping we don't see often. This is probably my general expectation in terms of how we pass methods into our composable functions https://searchfox.org/mozilla-mobile/source/fenix/app/src/main/java/org/mozilla/fenix/home/recenttabs/view/RecentTabViewHolder.kt#41-44.

Reply:
Yeah I would never suggest we intentionally use trailing lambdas the way I demonstrated in that middle case. Just pointing out that by putting functional parameters in the middle of the list we could also never accidentally use them in the middle case way. Compose tends to favor using named parameters anyway, so it should be an easily avoidable hypothetical. 

I'll update these signatures

Reply:
This is a great discussion. Thanks for updating the signatures. Ideally, we try to converge towards a common Fenix/team style for writing composable functions similar message to https://github.com/mozilla-mobile/fenix/pull/22953#discussion_r774230937 - that way it's easier for others to follow and replicate as a standard convention. For the most part, I have been looking at the compose sample apps and even digging into the compose material UI implementation to just try and absorb what is the best practise, but with anything that's a work in progress short of someone writing a linter that enforces an opinionated and accepted style.